Using an empty parameter set causes an error

Example:

```python
import pytest_cases


@pytest_cases.parametrize(x=[])
def case_foo(x):
    return x


@pytest_cases.parametrize_with_cases("x", case_foo)
def test_foo(x):
    pass
```

Note the empty argvalue list. (In practice this will be some dynamically computed value, e.g. a set intersection of dynamically computed sets that results in an empty set).

Traceback:

<details>

```none
============================================================================================== test session starts ==============================================================================================
platform win32 -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: C:\Users\Paolo\AppData\Roaming\JetBrains\PyCharm2020.3\scratches
plugins: cases-2.7.1
collected 0 items / 1 error                                                                                                                                                                                      

==================================================================================================== ERRORS =====================================================================================================
__________________________________________________________________________________________ ERROR collecting scratch.py __________________________________________________________________________________________
scratch.py:10: in <module>
    def test_foo(x):
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\common_pytest.py:786: in apply
    return apply_decorator(test_or_fixture_func, container)
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\case_parametrizer_new.py:119: in _apply_parametrization
    argvalues = get_parametrize_args(host_class_or_module, cases_funs, debug=debug)
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\case_parametrizer_new.py:274: in get_parametrize_args
    return [c for _f in cases_funs for c in case_to_argvalues(host_class_or_module, _f, debug)]
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\case_parametrizer_new.py:274: in <listcomp>
    return [c for _f in cases_funs for c in case_to_argvalues(host_class_or_module, _f, debug)]
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\case_parametrizer_new.py:300: in case_to_argvalues
    meta = MiniMetafunc(case_fun)
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\common_pytest.py:562: in __init__
    self.update_callspecs()
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\pytest_cases\common_pytest.py:586: in update_callspecs
    self.parametrize(argnames=pmark.param_names, argvalues=argvals, ids=pmark.param_ids,
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\_pytest\python.py:1035: in parametrize
    argnames, parameters = ParameterSet._for_parametrize(
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\_pytest\mark\structures.py:196: in _for_parametrize
    mark = get_empty_parameterset_mark(config, argnames, func)
c:\users\paolo\code\jb-pycharmprojects\loveletter\venv\lib\site-packages\_pytest\mark\structures.py:61: in get_empty_parameterset_mark
    requested_mark = config.getini(EMPTY_PARAMETERSET_OPTION)
E   AttributeError: 'NoneType' object has no attribute 'getini'
============================================================================================ short test summary info ============================================================================================
ERROR scratch.py - AttributeError: 'NoneType' object has no attribute 'getini'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

</details>

This is caused by setting `config` in `MiniMetafunc` to `None`:

https://github.com/smarie/python-pytest-cases/blob/8febdeeb52f6dcfcb54aa1e8e7d966427d030ad5/pytest_cases/common_pytest.py#L546-L549

I don't know the details of how this works, but this should probably be given the [`pytest.config.Config`](https://docs.pytest.org/en/stable/reference.html#pytest.config.Config) object.

--- a/pytest_cases/common_pytest.py
+++ b/pytest_cases/common_pytest.py
@@ -4,6 +4,8 @@
 # License: 3-clause BSD, <https://github.com/smarie/python-pytest-cases/blob/master/LICENSE>
 from __future__ import division
 
+import sys
+
 from makefun import add_signature_parameters, wraps
 
 try:  # python 3.3+
@@ -537,8 +539,6 @@ def mini_idvalset(argnames, argvalues, idx):
 try:
     from _pytest.compat import getfuncargnames  # noqa
 except ImportError:
-    import sys
-
     def num_mock_patch_args(function):
         """ return number of arguments used up by mock arguments (if any) """
         patchings = getattr(function, "patchings", None)
@@ -588,7 +588,9 @@ class MiniFuncDef(object):
 class MiniMetafunc(Metafunc):
     # noinspection PyMissingConstructor
     def __init__(self, func):
-        self.config = None
+        from .plugin import PYTEST_CONFIG  # late import to ensure config has been loaded by now
+
+        self.config = PYTEST_CONFIG
         self.function = func
         self.definition = MiniFuncDef(func.__name__)
         self._calls = []

--- a/pytest_cases/plugin.py
+++ b/pytest_cases/plugin.py
@@ -1290,14 +1290,21 @@ def pytest_addoption(parser):
     )
 
 
+# will be loaded when the pytest_configure hook below is called
+PYTEST_CONFIG = None  # type: _pytest.config.Config
+
+
 # @hookspec(historic=True)
 def pytest_configure(config):
+    global PYTEST_CONFIG
     # validate the config
     allowed_values = ('normal', 'skip')
     reordering_choice = config.getoption(_OPTION_NAME)
     if reordering_choice not in allowed_values:
         raise ValueError("[pytest-cases] Wrong --%s option: %s. Allowed values: %s"
                          "" % (_OPTION_NAME, reordering_choice, allowed_values))
+    # store the received config object for future use; see #165 & #166
+    PYTEST_CONFIG = config
 
 
 @pytest.hookimpl(tryfirst=True, hookwrapper=True)

--- a/pytest_cases/tests/cases/issues/test_issue_165.py
+++ b/pytest_cases/tests/cases/issues/test_issue_165.py
@@ -1,0 +1,11 @@
+import pytest_cases
+
+
+@pytest_cases.parametrize(x=[])
+def case_empty(x):
+    return x  # pragma: no cover
+
+
+@pytest_cases.parametrize_with_cases("x", case_empty)
+def test_empty_parameter_set(x):
+    assert False  # pragma: no cover

